### PR TITLE
Introduce stream-chat-android-benchmark module and add baseline profiles

### PR DESCRIPTION
**Pull request by skydoves**
### 🎯 Goal

Introduce `stream-chat-android-benchmark` module and add baseline profiles for improving runtime startup performance in Jetpack Compose & clients.

### 🛠 Implementation details

- Added the `stream-chat-android-benchmark` module.
- Added generated baseline profiles for the compose & client module

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Assigned a person / code owner group (required)
- [x] PR targets the `develop` branch

#### Code & documentation
- [x] Changelog is updated with client-facing changes
- [x] New code is covered by unit tests

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### CODE REVIEW

File: buildSrc/src/main/kotlin/io/getstream/chat/android/Dependencies.kt [hunk 1]
```diff
@@ -76,6 +76,10 @@ object Versions {
     internal const val TIMBER = "5.0.1"
     internal const val TURBINE = "0.12.1"
     internal const val WORK = "2.7.1"
+    internal const val ANDROIDX_TEST = "1.4.0"
+    internal const val BASE_PROFILE = "1.2.0"
+    internal const val MACRO_BENCHMARK = "1.1.0"
+    internal const val ANDROIDX_UI_AUTOMATOR = "2.2.0"
```
Comment (JcMinarro) [inline buildSrc/src/main/kotlin/io/getstream/chat/android/Dependencies.kt:82]: Could we sort them alphabetically?

File: buildSrc/src/main/kotlin/io/getstream/chat/android/Dependencies.kt [hunk 2]
```diff
@@ -211,7 +215,13 @@ object Dependencies {
...
     const val workRuntimeKtx = "androidx.work:work-runtime-ktx:${Versions.WORK}"
     const val workTesting = "androidx.work:work-testing:${Versions.WORK}"
-
+    const val baseProfile =
+        "androidx.profileinstaller:profileinstaller:${Versions.BASE_PROFILE}"
+    const val macroBenchmark =
+        "androidx.benchmark:benchmark-macro-junit4:${Versions.MACRO_BENCHMARK}"
+    const val uiAutomator =
+        "androidx.test.uiautomator:uiautomator:${Versions.ANDROIDX_UI_AUTOMATOR}"
+    const val testRunner = "androidx.test:runner:${Versions.ANDROIDX_TEST}"
```
Comment (JcMinarro) [inline buildSrc/src/main/kotlin/io/getstream/chat/android/Dependencies.kt:224]: Same here

File: buildSrc/src/main/kotlin/io/getstream/chat/android/Dependencies.kt (hunk 1 above)
Comment (skydoves) [reply buildSrc/src/main/kotlin/io/getstream/chat/android/Dependencies.kt]: Done: [3110edd](https://github.com/GetStream/stream-chat-android/pull/4610/commits/3110eddc3a7e82fb85397a13d91d77662a26a226)

File: buildSrc/src/main/kotlin/io/getstream/chat/android/Dependencies.kt (hunk 2 above)
Comment (skydoves) [reply buildSrc/src/main/kotlin/io/getstream/chat/android/Dependencies.kt]: Done: [3110edd](https://github.com/GetStream/stream-chat-android/pull/4610/commits/3110eddc3a7e82fb85397a13d91d77662a26a226)

**Review by filbabic — APPROVED** [review]

**Pull request merged**